src/components/AllocationOverview.js

```diff
@@ -0,0 +1,276 @@
+        errorDeletedPayment
+    }] = useMutation(DELETE_ALLOCATION, {
+        variables: {
+            id: allocationInfo.id
+        },
+        refetchQueries: [{
+            query: GET_ALLOCATIONS,
+            variables: {
+                projectId: contributorAllocation ? contributorAllocation.project.id : null,
+                contributorId: contributorAllocation ? contributorAllocation.contributor.id : null
+            }
+        }, {
+            query: GET_PROJECT_CONTRIBUTORS,
+            variables: {
+                id: contributorAllocation ? contributorAllocation.project.id : null
+            }
+        }]
+    })
+
+    const [updateAllocation, {
+        dataUpdatedAllocation,
+        loadingUpdatedAllocation,
+        errorUpdatedAllocation
+    }] = useMutation(UPDATE_ALLOCATION, {
+        refetchQueries: [{
+            query: GET_CONTRIBUTOR_ALLOCATIONS,
+            variables: {
+                id: contributorAllocation ? contributorAllocation.contributor.id : null
+            }
+        }]
+    })
+
+    useEffect(() => {
+        if (contributorAllocation) {
+            getClientPayments({
+                variables: {
+                    clientId: contributorAllocation.project.client.id
+                }
+            })
+            getContributorRates({
+                variables: {
+                    id: contributorAllocation.contributor.id
+                }
+            })
+            setUpdatedAllocationEndDate(moment(allocation.end_date, 'x')['_d'])
+            setUpdatedAllocationPayment(contributorAllocation.payment)
+            setUpdatedAllocationStartDate(moment(allocation.start_date, 'x')['_d'])
+        }
+    }, [contributorAllocation])
+
+    const handleDeleteAllocation = async () => {
+        const paymentDeleted = await deleteAllocation()
+        onClose()
+    }
+
+    const handleUpdateAllocation = async ({
+        allocation,
+        contributor,
+        contributorRates,
+        endDate,
+        payment,
+        rate,
+        startDate,
+    }) => {
+        //look for rate with same values
+        const selectedRate = {}
+        const existingRate = findKey(
+            contributorRates,
+            {
+                'hourly_rate': rate.hourly_rate.toString(),
+                'total_expected_hours': Number(rate.total_expected_hours),
+                'type': rate.type,
+                'currency': selectedCurrency
+            }
+        )
+        if (existingRate) {
+            selectedRate.id = contributorRates[existingRate].id
+        } else {
+            //create rate
+            const newRate = await createRate({
+                variables: {
+                    hourly_rate: rate.hourly_rate.toString(),
+                    total_expected_hours: Number(rate.total_expected_hours),
+                    type: rate.type,
+                    currency: selectedCurrency,
+                    contributor_id: contributor.id
+                }
```

Comment:
Just noting the inconsistency here:

```
{
    'hourly_rate': ...
}

{
    hourly_rate: ...
}
```

I would prefer without single quotes there's really no need for them